### PR TITLE
add donate banner, make close button a mode of button component

### PR DESCRIPTION
**Pull request by benjiwheeler**
### Resolves:

Resolves https://github.com/LLK/scratch-www/issues/5318

### Changes:

* adds donate banner component and css
* modifies button component to have "close" version, to be reusable
* modifies splash/splash.jsx and splash/presentation.jsx to collect similar banner handling in one place
* adds handling for closing the banner and for redirecting user to donate page

### Issues

Closing the banner doesn't persist for long -- if you browse away from the splash screen and come back, it will reappear.

### Test Coverage:

unclear if it makes sense to add tests?

### Screenshots

image

image

image

image

image

image

### CODE REVIEW

**Comment by benjiwheeler** [comment]
next step: go over with @kmlshalf

**Comment by GrahamSH-LLK** [comment]
I assume that this PR didn't cause this?
image
If it didn't cause it, should I open an issue?

**Comment by RedGuy12** [comment]
> I assume that this PR didn't cause this?
> image
> If it didn't cause it, should I open an issue?

@GrahamSH-LLK i think that's because he's running it locally

File: src/views/splash/donate/donate-banner.scss [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    z-index: 8;
+    background-color: $ui-aqua-dark;
+    padding: 0;
+    overflow: hidden;
+    align-items: center;
+    justify-content: center;
+
+    .donate-container {
+        display: flex;
+        // width: $desktop;
```
Comment (kchadha) [inline src/views/splash/donate/donate-banner.scss:18]: cruft?

File: src/views/splash/donate/donate-banner.scss [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        right: 1rem;
+        top: auto;
+    }
+}
+
+// @media only screen and (min-width: $mobile) and (max-width: $mobileIntermediate) {
+//     .donate-banner .donate-container .donate-button {
+//         margin: 0 0 1rem 0;
+//     }
+// }
```
Comment (kchadha) [inline src/views/splash/donate/donate-banner.scss:70]: cruft?

File: src/views/splash/donate/donate-banner.scss (hunk 1 above)
Comment (kchadha) [reply src/views/splash/donate/donate-banner.scss:18]: and the two commented out `justify-content` lines below

**Review by kchadha — COMMENTED** [review]
There's some commented out code.

In testing, there is a moment while resizing my browser window, where the close button overlaps with the donate button which isn't ideal.

Other than those two things, the rest of these changes LGTM.

File: src/views/splash/donate/donate-banner.scss (hunk 2 above)
Comment (benjiwheeler) [reply src/views/splash/donate/donate-banner.scss:70]: Removed

**Comment by benjiwheeler** [comment]
Updated, please take a look!

**Review by kchadha — APPROVED** [review]
LGTM! Thanks for making the changes =]

**Pull request merged**